### PR TITLE
[CSL-481] Update constructorio-node to support `patchdelta`

### PR DESCRIPTION
**Pull request by esezen**

### CODE REVIEW

File: src/modules/catalog.js [hunk 1]
```diff
@@ -1584,6 +1584,109 @@ class Catalog {
...
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/full_catalog
+   */
+  async patchCatalog(parameters = {}) {
+    let requestUrl;
+    const queryParams = { patch_delta: true };
+    const formData = new FormData();
+    const fetch = (this.options && this.options.fetch) || nodeFetch;
+
+    if (parameters) {
```
Comment (esezen) [inline src/modules/catalog.js:1607]: Next 60 lines are shared between `replaceCatalog`, `updateCatalog`, and `patchCatalog`. Would it make sense to extract this piece to a helper function in this PR on in another one?

File: src/modules/catalog.js [hunk 2]
```diff
@@ -1584,6 +1584,109 @@ class Catalog {
...
   }
 
+  /**
+   * Send patch delta catalog files to patch the current catalog
+   *
+   * @function patchCatalog
+   * @param {object} parameters - Additional parameters for catalog details
+   * @param {string} parameters.section - The section that you want to update
+   * @param {string} [parameters.notification_email] - An email address where you'd like to receive an email notifcation in case the task fails.
+   * @param {boolean} [parameters.force] - Process the catalog even if it will invalidate a large number of existing items. Defaults to False.
```
Comment (sblaurock) [inline src/modules/catalog.js:1594]: We can specify defaults using JSDoc: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

File: src/modules/catalog.js [hunk 3]
```diff
@@ -1584,6 +1584,109 @@ class Catalog {
     }
   }
 
+  /**
+   * Send patch delta catalog files to patch the current catalog
+   *
+   * @function patchCatalog
+   * @param {object} parameters - Additional parameters for catalog details
+   * @param {string} parameters.section - The section that you want to update
+   * @param {string} [parameters.notification_email] - An email address where you'd like to receive an email notifcation in case the task fails.
```
Comment (sblaurock) [inline src/modules/catalog.js:1593]: Remove period at end of sentence as none of the other params use it

File: src/modules/catalog.js (hunk 1 above)
Comment (sblaurock) [reply src/modules/catalog.js:1607]: I'm happy to ship this as is and do the refactor in a separate PR - big fan of incremental updates 😄 

Want to throw a story on the board?

**Review by sblaurock — COMMENTED** [review]
This looks great! A few comments, but it's close.

File: src/modules/catalog.js (hunk 2 above)
Comment (esezen) [reply src/modules/catalog.js:1594]: Sounds good. Also added the default for `force` under `replaceCatalog` and `updateCatalog`

File: src/modules/catalog.js (hunk 3 above)
Comment (esezen) [reply src/modules/catalog.js:1593]: Sorry copy-paste. Also removed from `updateCatalog`

File: src/modules/catalog.js (hunk 1 above)
Comment (esezen) [reply src/modules/catalog.js:1607]: Done!

**Review by sblaurock — APPROVED** [review]
Looking great - nice work.

**Pull request merged**